src/ucs/datastruct/ptr_array.h

```diff
@@ -184,6 +201,44 @@ ucs_ptr_array_is_empty(ucs_ptr_array_t *ptr_array)
 }
 
 
+/**
+ * Shift an input value so it will fit in a ptr array slot.
+ *
+ * @param [in] new_val        Value to be put into a slot (at some point).
+ *
+ * @return The same value, as it would be stored in the ptr array.
+ */
+static UCS_F_ALWAYS_INLINE uintptr_t ucs_ptr_array_shift(uintptr_t new_val)
+{
+    return new_val << UCS_PTR_ARRAY_NEXT_SHIFT;
+}
+
+
+/**
+ * Shift back the value retrieved from a ptr array (@ref ucs_ptr_array_shift).
+ *
+ * @param [in] old_val        Value which has been stored in the ptr array slot.
+ *
+ * @return The original value stored in the ptr array slot (before shifting).
+ */
+static UCS_F_ALWAYS_INLINE uintptr_t ucs_ptr_array_unshift(uintptr_t old_val)
+{
+    return old_val >> UCS_PTR_ARRAY_NEXT_SHIFT;
+}
```

Comment:
seems unused

Reply:
yes, it is unused. It is to be used later, as part of a larger change: #6783 (from stats.h).

Reply:
so pls move to this next PR

Reply:
fair enough, will do.